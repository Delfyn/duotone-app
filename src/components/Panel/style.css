.panel {
  background-color: #fff;
  box-shadow: 0 0 12px 0 rgba(0, 0, 0, 0.1), 0 10px 30px 0 rgba(0, 0, 0, 0.2);
  border-radius: .5rem;
  position: fixed;
  z-index: 3;
  color: var(--text-color);
  min-width: 18rem;
  max-width: 20rem;
}

.panel-draggable {
  cursor: move;
  height: 1rem;
}

.panel main {
  border-width: 1px;
  border-color: var(--border-color);
  border-top-style: solid;
  border-bottom-style: solid;
  font-size: 1rem;
}
