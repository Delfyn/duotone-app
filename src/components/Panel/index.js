/* @flow */

import React, { Children, cloneElement } from 'react';
import Draggable from 'react-draggable';
import './style.css';

const Panel = (props: any) => {
  const childProps = Object.assign({}, props);
  delete childProps.children;
  const childrenWithProps = Children.map(props.children, child =>
    cloneElement(child, {
      ...childProps,
    }),
  );

  return (
    <Draggable handle=".panel-draggable" allowAnyClick={false}>
      <div className="panel mt4 ml3">
        <div className="panel-draggable" />
        {childrenWithProps}
      </div>
    </Draggable>
  );
};

export default Panel;
