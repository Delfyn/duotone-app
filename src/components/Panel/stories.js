/* @flow */

import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Panel from './index';

storiesOf('Panel', module)
  .add('Simple example', () => (
    <Panel>
      <h1>Panel Headline</h1>
      <p>You can drag me around if you grab me at the top.</p>
    </Panel>
  ));
