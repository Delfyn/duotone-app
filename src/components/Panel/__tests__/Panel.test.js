import React from 'react';
import renderer from 'react-test-renderer';
import Panel from '../index';

test('Panel renders correctly', () => {
  const component = renderer.create(<Panel />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
