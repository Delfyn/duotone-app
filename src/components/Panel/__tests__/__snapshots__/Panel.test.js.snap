// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly 1`] = `
<div
  className="panel mt4 ml3 react-draggable"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "touchAction": "none",
      "transform": "translate(0px,0px)",
    }
  }
  transform={null}
>
  <div
    className="panel-draggable"
  />
</div>
`;
